### PR TITLE
Release/4.3.0

### PR DESCRIPTION
**Pull request by luu-alex**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../CONTRIBUTIONS.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-core/CHANGELOG.md [hunk 1]
```diff
@@ -201,8 +201,10 @@ Documentation:
 
 -   Added `isMetaMaskProvider` function to check if provider is metamask (#6534)
 
-## [Unreleased]
+## [4.4.0]
```
Comment (jdevcs) [inline packages/web3-core/CHANGELOG.md]: Its a patch fix for following change so should it be 4.3.2

File: packages/web3-eth-accounts/CHANGELOG.md [hunk 2]
```diff
@@ -149,4 +149,8 @@ Documentation:
 
 -   Fixed `recover` function, `v` will be normalized to value 0,1 (#6344) 
 
+### Changed
```
Comment (jdevcs) [inline packages/web3-eth-accounts/CHANGELOG.md]: if accounts version is not bumped there shouldn't be any change in its file

File: packages/web3-eth-contract/CHANGELOG.md [hunk 3]
```diff
@@ -347,8 +347,10 @@ Documentation:
...
 -	Will populate `data` for transactions in contract for metamask provider instead of `input` (#6534)
 
-## [Unreleased]
+## [4.2.0]
 
 ### Changed
 
--	`contractDataInputFill` has been defaulted to `data`, istead of `input`. (#6622)
\ No newline at end of file
+-	By default, contracts will fill `data` instead of `input` within method calls (#6622)
```
Comment (jdevcs) [inline packages/web3-eth-contract/CHANGELOG.md:354]: its also patch fix so should be 4.1.4

File: packages/web3-eth-personal/CHANGELOG.md [hunk 4]
```diff
@@ -147,4 +147,10 @@ Documentation:
 
 -   Dependencies updated
 
+## [4.0.9]
+
+### Changed
+
+-   Dependencies updated
```
Comment (jdevcs) [inline packages/web3-eth-personal/CHANGELOG.md]: if there are only patch & minor dep updates we can skip changes in these packages,

File: packages/web3-validator/package.json [hunk 5]
```diff
@@ -47,8 +47,8 @@
 	"dependencies": {
 		"ethereum-cryptography": "^2.0.0",
 		"util": "^0.12.5",
-		"web3-errors": "^1.1.3",
-		"web3-types": "^1.3.0",
+		"web3-errors": "^1.1.4",
```
Comment (jdevcs) [inline packages/web3-validator/package.json]: same here and in other packages

**Comment by jdevcs** [comment]
also could you see reason of failing integration tests in geth

File: packages/web3-core/CHANGELOG.md (hunk 1 above)
Comment (luu-alex) [reply packages/web3-core/CHANGELOG.md]: updated to 4.3.2

File: packages/web3-eth-accounts/CHANGELOG.md (hunk 2 above)
Comment (luu-alex) [reply packages/web3-eth-accounts/CHANGELOG.md]: changed it back

File: packages/web3-eth-contract/CHANGELOG.md (hunk 3 above)
Comment (luu-alex) [reply packages/web3-eth-contract/CHANGELOG.md:354]: changed to 4.1.4

**Comment by luu-alex** [comment]
sepolia, mainnet and geth estimategas function tests are failing. This has been occuring for mainnet and sepolia for a while but is quite new for geth estimate_gas function to be failing

File: packages/web3-eth-personal/CHANGELOG.md (hunk 4 above)
Comment (luu-alex) [reply packages/web3-eth-personal/CHANGELOG.md]: skipped packages that are only patch/minor dep updates

**Comment by Muhammad-Altabba** [comment]
> also could you see reason of failing integration tests in geth

I think `geth` has been updated; And with that update the gas estimate differently. But this probably has to be more investigated.

File: CHANGELOG.md [hunk 6]
```diff
@@ -2237,6 +2238,12 @@ If there are any bugs, improvements, optimizations or any new feature proposal f
 -   Interface `MetaMaskProvider` added and is part of `SupportedProviders` (#6534)
 -   `gasPrice` was added to `Transaction1559UnsignedAPI` type. (#6539)
 
+#### web3-eth-accounts
+
+-   Added public function `privateKeyToPublicKey`
+-   Added exporting `BaseTransaction` from the package (#6493)
+-   Added exporting `txUtils` from the package (#6493)
+
```
Comment (Muhammad-Altabba) [inline CHANGELOG.md]: Those are already mentioned in this file under `[4.1.0]`

```suggestion
```

**Review by jdevcs — APPROVED** [review]

**Review by Muhammad-Altabba — COMMENTED** [review]
Could you please double check the errors. There is this error in the unit lint, for example:

> web3-core:    27:1  error    'web3-providers-ws' should be listed in the project's dependencies. Run 'npm i -S web3-providers-ws' to add it  import/no-extraneous-dependencies


And this at the [E2E Network Tests (http, mainnet)](https://github.com/web3/web3.js/actions/runs/7048626658/job/19185351328?pr=6630#logs):

 FAIL  test/e2e/get_proof.test.ts
  ● sepolia tests - getProof › getProof

    expect(received).toMatchObject(expected)

    - Expected  - 1
    + Received  + 0

    @@ -1,7 +1,6 @@
      Object {
    -   "accountProof": Any<Array>,
        "balance": Any<BigInt>,
        "codeHash": Any<String>,
        "nonce": Any<BigInt>,
        "storageHash": Any<String>,
        "storageProof": Any<Array>,

      70 | 			expect(result).toEqual(expectedProof);
      71 | 		} else {
    > 72 | 			expect(result).toMatchObject<AccountObject>({
         | 			               ^
      73 | 				accountProof: expect.any(Array<string>),
      74 | 				balance: expect.any(BigInt),
      75 | 				codeHash: expect.any(String),

      at test/e2e/get_proof.test.ts:72:19
      at fulfilled (test/e2e/get_proof.test.ts:22:58)

**Review by avkos — APPROVED** [review]

**Comment by luu-alex** [comment]
@Muhammad-Altabba  fixed both, thank you

**Review by Muhammad-Altabba — APPROVED** [review]
Could you please create an issue for the failing tests: https://github.com/web3/web3.js/actions/runs/7091989375/job/19302456686?pr=6630
Many thanks,

**Pull request merged**